src/components/calcite-action/calcite-action.tsx

```diff
@@ -51,10 +51,15 @@ export class CalciteAction {
    */
   @Prop() text: string;
 
+  /**
+   * @deprecated Use 'textMode' instead.
+   */
+  @Prop() textEnabled = false;
+
   /**
    * Indicates whether the text is displayed.
    */
-  @Prop({ reflect: true }) textEnabled = false;
+  @Prop({ reflect: true }) textMode: "show" | "hide" | "auto" = "hide";
```

Comment:
Modes should probably be moved to a type.

Reply:
WDYT about `textDisplay = "visible" | "hidden" | "on-interaction"`? I'm sure there's a better word for that last one. 😄

Reply:
@jcfranco Yeah I like `visible`, `hidden` much better.

How about just `interactive`?